src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala

```diff
@@ -49,33 +51,52 @@ trait RestJsonClient extends FireCloudRequestBuilding {
-    requestToObject(true, req, compressed, useFireCloudHeader, connector)
+  def authedRequestToObject[T](req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None, label: Option[String] = None)(implicit userInfo: WithAccessToken, unmarshaller: Unmarshaller[T], ers: ErrorReportSource): Future[T] = {
+    requestToObject(true, req, compressed, useFireCloudHeader, connector, label)
   }
 
-  def unAuthedRequestToObject[T](req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None)(implicit unmarshaller: Unmarshaller[T], ers: ErrorReportSource): Future[T] = {
+  def unAuthedRequestToObject[T](req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None, label: Option[String] = None)(implicit unmarshaller: Unmarshaller[T], ers: ErrorReportSource): Future[T] = {
     implicit val userInfo:WithAccessToken = null
-    requestToObject(false, req, compressed, useFireCloudHeader, connector)
+    requestToObject(false, req, compressed, useFireCloudHeader, connector, label)
   }
 
+  // zero usages in the codebase
   def adminAuthedRequestToObject[T](req:HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false)(implicit unmarshaller: Unmarshaller[T], ers: ErrorReportSource): Future[T] = {
     resultsToObject(adminAuthedRequest(req, compressed, useFireCloudHeader))
   }
 
-  def requestToObject[T](auth: Boolean, req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None)(implicit userInfo: WithAccessToken, unmarshaller: Unmarshaller[T], ers: ErrorReportSource): Future[T] = {
+  private def requestToObject[T](auth: Boolean, req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false, connector: Option[ActorRef] = None, label: Option[String] = None)(implicit userInfo: WithAccessToken, unmarshaller: Unmarshaller[T], ers: ErrorReportSource): Future[T] = {
+    val tick = if (label.nonEmpty) System.currentTimeMillis() else NoPerfLabel
+
     val resp = if(auth) {
       userAuthedRequest(req, compressed, useFireCloudHeader, connector)
     } else {
       unAuthedRequest(req, compressed, useFireCloudHeader, connector)
     }
-    resultsToObject(resp)
+    resultsToObject(resp, label, tick)
   }
 
-  def resultsToObject[T](resp: Future[HttpResponse])(implicit unmarshaller: Unmarshaller[T], ers: ErrorReportSource): Future[T] = {
+  private def resultsToObject[T](resp: Future[HttpResponse], label: Option[String] = None, tick: Long = NoPerfLabel)(implicit unmarshaller: Unmarshaller[T], ers: ErrorReportSource): Future[T] = {
     resp map { response =>
+
+      if (label.nonEmpty && tick != NoPerfLabel) {
```

Comment:
Can
```scala
if (label.nonEmpty && tick != NoPerfLabel) {
```
be reduced to
```scala
if (tick != NoPerfLabel) {
```
?

Reply:
Here, I'd prefer to keep both checks. Since `label` and `tick` are both passed in as arguments to this method, a caller could omit one, and I think it's best to keep both checks.